### ISSUE TITLE
[Bug] Core UserCartRecalculationListener triggers order processing for any endpoint for authorized user, resulting in deadlocks

### ISSUE THREAD

**Issue by diimpp** [Sylius/ShopApiPlugin#681]
If `shop user` is logged in and have at least one previously assigned cart, then [CoreBundle/EventListener/UserCartRecalculationListener](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php) will trigger `orderProcessing` on **each** authorized request to **any** shop-api endpoint.

This happens due the fact that `ShopApiPlugin` doesn't define `CartContext` and `Core\CartContext` considers last assigned cart as current cart, which is not true for shop-api. It also expected be triggered only at login in html website, but in shop-api it happens in each request with JWT token.

This results in metric ton of deadlocks for any authorized user and renders SPA app unusable.

Solution is to override listener and disable it.

**Comment by mamazu** [comment]
And then we should just recalculate the cart on every request inside the `shop-api/checkout/` routes right?

**Comment by diimpp** [comment]
I don't think we should additionally recalculate cart in every request at cart or checkout as most of those endpoints already doing their own `order processing`.

Original sylius strategy doesn't quite fit shop-api stateless state and any new strategies will heavily depend on how frontend will use api endpoints. (For example new `/carts/{token}/refresh`, which can be called on login by frontend)

So I think best course of action is to just disable `Core\UserCartRecalculationListener` and don't do anything else.

/Offtopic,
in my opinion shop-api v1 is kind of done and not salvageable as it was released too early and only v2 with redesign will save it. (All contexts should be re-defined in shop-api, `Query` object should be introduces in addition to `Command`, etcetera)
So we should look more in maintenance, than new features.

For example I've implemented `out of stock validator` in my project, but it's not pretty and runs on modified version of shop-api and it would be plenty of work to make proper PR.
